plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -3342,4 +3348,59 @@ public void testDroppingIcebergAndCreatingANewTableWithTheSameNameShouldBePossib
         assertThat(query("SELECT min(a_varchar) FROM test_iceberg_recreate")).matches("VALUES CAST('Trino' AS varchar)");
         dropTable("test_iceberg_recreate");
     }
+
+    @Test
+    public void testPathHiddenColumn()
+    {
+        String tableName = "test_path_" + randomTableSuffix();
+        @Language("SQL") String createTable = "CREATE TABLE " + tableName + " " +
+                "WITH ( partitioning = ARRAY['col1'] ) AS " +
+                "SELECT * FROM (VALUES " +
+                "(0, 0), (3, 0), (6, 0), " +
+                "(1, 1), (4, 1), (7, 1), " +
+                "(2, 2), (5, 2) " +
+                " ) t(col0, col1) ";
+        assertUpdate(createTable, 8);
+        assertTrue(getQueryRunner().tableExists(getSession(), tableName));
+
+        TableMetadata tableMetadata = getTableMetadata(ICEBERG_CATALOG, TPCH_SCHEMA, tableName);
+
+        List<String> columnNames = ImmutableList.of("col0", "col1", PATH_COLUMN_NAME);
+        List<ColumnMetadata> columnMetadatas = tableMetadata.getColumns();
+        assertEquals(columnMetadatas.size(), columnNames.size());
+        for (int i = 0; i < columnMetadatas.size(); i++) {
+            ColumnMetadata columnMetadata = columnMetadatas.get(i);
+            assertEquals(columnMetadata.getName(), columnNames.get(i));
+            if (columnMetadata.getName().equals(PATH_COLUMN_NAME)) {
+                // $path should be hidden column
+                assertTrue(columnMetadata.isHidden());
+            }
+        }
+        int numberOfPartitions = computeActual(format("SELECT partition FROM \"%s$partitions\"", tableName)).getMaterializedRows().size();
+        assertEquals(numberOfPartitions, 3);
+
+        MaterializedResult results = computeActual(format("SELECT *, \"%s\" FROM %s", FILE_PATH.colName(), tableName));
+        Map<Integer, String> partitionPathMap = new HashMap<>();
+        for (int i = 0; i < results.getRowCount(); i++) {
+            MaterializedRow row = results.getMaterializedRows().get(i);
+            int col0 = (int) row.getField(0);
+            int col1 = (int) row.getField(1);
+            String pathName = (String) row.getField(2);
+            String parentDirectory = new org.apache.hadoop.fs.Path(pathName).getParent().toString();
+
+            assertTrue(pathName.length() > 0);
+            assertEquals(col0 % 3, col1);
+            if (partitionPathMap.containsKey(col1)) {
+                // the rows in the same partition should be in the same partition directory
+                assertEquals(partitionPathMap.get(col1), parentDirectory);
+            }
+            else {
+                partitionPathMap.put(col1, parentDirectory);
+            }
+        }
+        assertEquals(partitionPathMap.size(), 3);
```

Comment:
This looks very complicated.

I think it would be enough to assert that

```
SELECT DISTINCT "$path" FROM table
```

returns same result as

```
SELECT file_path FROM "table$files"
```

you can do this with 

```
assertThat(query(....))
  .matches(....)
```

Reply:
@findepi thanks.
I believe what you are suggesting, I have already added a similar test case, please check. https://github.com/trinodb/trino/blob/7936d9fa375240c47eeba7d267b3a3d8d1ae1b1d/plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java#L95-L99

And if this one is enough then I'm totally fine to avoid the `BaseIcebergConnectorTest.testPathHiddenColumn` 
@findinpath please suggest, if that is fine with you as well.

Reply:
modified @findepi